src/handle.rs

```diff
@@ -27,13 +27,25 @@ impl Drop for OwnedHandle {
         if Arc::strong_count(handle) == 1 {
             // This is the last reference to the arc
             let id = write_end.get_id_mut();
-            if let Ok(response) = write_end.send_close_request(id, Cow::Borrowed(handle)) {
-                // Requests is already added to write buffer, so wakeup
-                // the `flush_task`.
-                self.get_auxiliary().wakeup_flush_task();
-                tokio::spawn(async move {
-                    let _ = response.wait().await;
-                });
+            match write_end.send_close_request(id, Cow::Borrowed(handle)) {
+                Ok(response) => {
+                    // Requests is already added to write buffer, so wakeup
+                    // the `flush_task`.
+                    self.get_auxiliary().wakeup_flush_task();
+                    let future = response.wait();
```

Comment:
Explanation of why I'd like this to be out of the `async` block:

`response.wait()` is basically a no-op, it simply takes out the inner value of [`AwaitableStatus`] and wrap it with a corresponding [`AwaitableStatusFuture`], you can check its [impl here].

Currently, `rustc` isn't very good at optimizing moves in the future, it often results in the size of the `Future` blows out, becomes double of its size.

Plus, the more states the `Future`s have, the harder it is to optimize and take advantage of the niche.

P.S. You can include this as a comment.

[`AwaitableStatus`]: https://docs.rs/openssh-sftp-client-lowlevel/latest/openssh_sftp_client_lowlevel/struct.AwaitableStatus.html
[`AwaitableStatusFuture`]: https://docs.rs/openssh-sftp-client-lowlevel/latest/openssh_sftp_client_lowlevel/struct.AwaitableStatusFuture.html
[impl here]: https://docs.rs/openssh-sftp-client-lowlevel/latest/src/openssh_sftp_client_lowlevel/awaitables.rs.html#154